### PR TITLE
Add new release issue template

### PR DESCRIPTION
**Pull request by lparsons**
## Summary Change Description

Releasing a new version requires a number of steps. Having a checklist can help ensure the proper steps are followed.

## Review Notes

More [detailed docs on the process are in NPLC Admin docs](https://nplcadmindocs.princeton.edu/index.php/TraceBase#New_Release). This template is meant to be an easy to read checklist of steps.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [x] Added changes to *Unreleased* section of [`changelog.md`](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

**Review by hepcat72 — APPROVED** [review]
I like the idea of having a release issue template.  I have some suggestions, but nothing blocking.

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+name: New Release
+about: Create a new release of TraceBase
+title: ''
+labels: ''
+assignees: ''
+---
+
+<!-- markdownlint-disable-next-line first-line-heading -->
+* [ ] `git pull`
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md]: IMO, the determination of what to release should be done prior to creating this issue, but I take your point, I'll add a little text. I'd like to keep this a lean checklist, however, with the docs here: https://nplcadmindocs.princeton.edu/index.php/TraceBase#New_Release

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* [ ] Polish `changelog.md`
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md]: See docs for further explanation.

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+* [ ] `git push`
+* [ ] Ensure all tests pass
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md:13]: I don't consider that sufficient, the tests should pass on GitHub.

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+* [ ] `git tag -a [TAGNAME]`
+* [ ] `git push origin [TAGNAME]`
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md:15]: Added link to docs

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+* [ ] [Create a GitHub release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release)
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md:16]: Yes, this is a checklist item, follow the link for instructions.

File: .github/ISSUE_TEMPLATE/new_release.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+* [ ] Add *Unreleased* to `changelog.md`
+* [ ] `git push`
+  * [ ] If release branch - merge into main and push
```
[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md]: Thanks, I'll try to make this clearer.

**Review by lparsons — COMMENTED** [review]
Thanks for the review @hepcat72, some good suggestions. I've made a few changes here and clarified the docs at: https://nplcadmindocs.princeton.edu/index.php/TraceBase#New_Release

**Comment by lparsons** [comment]
@fkang-pu and @jcmatese - Just wanted to make sure you two were able to take a look at this, of course we can update later as people go through the process and refine things.

File: .github/ISSUE_TEMPLATE/new_release.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply .github/ISSUE_TEMPLATE/new_release.md]: Perhaps link this item to that section of the docs.

[the remark this answers is not in the document]
Comment (lparsons) [reply .github/ISSUE_TEMPLATE/new_release.md]: There is a link at the bottom of the checklist to a section in the docs that mirrors this checklist with additional detail. Perhaps we could move it to the top to make it more obvious is exists?

File: .github/ISSUE_TEMPLATE/new_release.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply .github/ISSUE_TEMPLATE/new_release.md:13]: That's what was unclear to me (when/where the tests should be ensured to pass).  All of the other directives (before and after) are operations that happen on a developer's workstation.  The only way it seemed clear to me was to specify that these were local tests, but I see that you should ensure they pass (on github) before creating the tag, so I would suggest:
```suggestion
* [ ] Ensure all tests pass on github
```

File: .github/ISSUE_TEMPLATE/new_release.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply .github/ISSUE_TEMPLATE/new_release.md:16]: Thanks for the clarification.  The link alone seemed redundant.  Without context to say what is to be done with the link, the link text comes across (to me) as just the make of the page, with makes it seem like it's either a reference/informational, i.e. "This checklist is based on this page".  I think making this read like a directive would disambiguate what the link is for.
```suggestion
* [ ] Follow steps on [Create a GitHub release](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release)
```

**Review by fkang-pu — APPROVED** [review]
Nice!

**Review by jcmatese — APPROVED** [review]

**Pull request merged**